src/steamship/agents/agents.py

```diff
@@ -0,0 +1,187 @@
+import uuid
+from abc import ABC, abstractmethod
+from typing import Any, List, Optional
+
+from pydantic import BaseModel
+
+from steamship import Block, File, Steamship, Tag, Task
+from steamship.invocable import InvocableResponse, PackageService, post
+from steamship.utils.kv_store import KeyValueStore
+
+
+class Action(BaseModel):
+    name: str
+    data: Any
+
+
+class FinishAction(Action):
+    pass
+
+
+class AgentContext(BaseModel, ABC):
+    # todo: define what should be in this context
+    # some set of state-related to agent (maybe request ids, callbacks, etc.)
```

Comment:
should this include a "transaction" log of sorts?

Reply:
@eob No, I think Files are the obvious place to be storing this context.

I think this context object is a very useful abstraction.  I wonder if remembering things (memory) is really a part of a context object and not a separate Tool.